### PR TITLE
Read config to set UI in readonly mode

### PR DESCRIPTION
**Pull request by Nexucis**
Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Comment by LukeTillman** [comment]
Thanks for taking a stab at this, @Nexucis. I think I want to understand the use case here a bit more. Why do you see someone making Perses read-only? Is this for folks who want to use source control to manage changes to dashboards?

**Comment by Nexucis** [comment]
@LukeTillman yes exactly. It would happen in case you have the dashboard in Git and then you deploy them in a storage volume read by Perses.

The deployment process could use the CRD (Custom Resource Document) in kubernetes when it will be available.

**Comment by Nexucis** [comment]
So in that configuration, you don't want to see someone overriding your dashboard without passing through a PR first

**Comment by eunicorn** [comment]
@Nexucis Rather than removing the edit button completely, what do you think about disabling save button instead? Users might still want to do some visual editing to update the json and then they can copy the json to open a PR in git

**Comment by Nexucis** [comment]
ah yeah that could happen indeed. But then we would need an export button no ? 

Also we might want to disable the dialog that is opened when we move out of the dashboard view and we have on going changes not saved. 

Just thinking widely, but having an `edit` button and then a disable `save` button would be weird no ?

**Comment by eunicorn** [comment]
Yeah we can implement an export button. This should be very easy to implement since @cndonovan already implemented it in our chronosphere product. Eventually we also want to add a json editor and users can copy the json from there as well.

> Just thinking widely, but having an `edit` button and then a disable `save` button would be weird no ?

agreed, we can show a banner that distinguishes that this is a different editing mode

**Comment by eunicorn** [comment]
@Nexucis What do you think about this? We can disable the `save` button and have a message next to it saying `Dashboard managed via code only. Download JSON and commit changes to Save.`

image

**Comment by Nexucis** [comment]
that sounds really cool @eunicorn !!

**Comment by Nexucis** [comment]
In the current code, the `export` button would export the non changed dashboard. Because you would need to save the dashboard to change the state of the current dashboard.

So we can't currently use the `export` button to get the modified dashboard :(

**Comment by eunicorn** [comment]
> In the current code, the `export` button would export the non changed dashboard. Because you would need to save the dashboard to change the state of the current dashboard.
> 
> So we can't currently use the `export` button to get the modified dashboard :(

Oops I forgot to also add the download button when in edit mode! This PR should fix that -https://github.com/perses/perses/pull/801

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - [FEATURE] Add ability to refresh dashboard #777
 - [FEATURE] Add a migration page #774
+- [ENHANCEMENT] Dashboard view can be configured to be readonly
```
Comment (eunicorn) [inline CHANGELOG.md:7]: missing the pull request number here

**Review by eunicorn — APPROVED** [review]
awesome thanks! :)

**Pull request merged**